<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><link rel="canonical" href="https://neurobioinfo.github.io/bpds/" />
      <link rel="shortcut icon" href="../img/favicon.ico" />
    <title>Best Practices for Data Storage - NeuroVIP</title>
    <link rel="stylesheet" href="../css/theme.css" />
    <link rel="stylesheet" href="../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Best Practices for Data Storage";
        var mkdocs_page_input_path = "bpds.md";
        var mkdocs_page_url = "/bpds/";
      </script>
    
    <!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href=".." class="icon icon-home"> NeuroVIP
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../getting_started/">Getting Started</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../upload_data/">Upload data</a>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../redundant_backups/">Redundant Backups</a>
                </li>
              </ul>
              <ul class="current">
                <li class="toctree-l1 current"><a class="reference internal current" href="#">Best Practices for Data Storage</a>
    <ul class="current">
    </ul>
                </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../download_data/">Download</a>
                </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="..">NeuroVIP</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href=".." class="icon icon-home" aria-label="Docs"></a></li>
      <li class="breadcrumb-item active">Best Practices for Data Storage</li>
    <li class="wy-breadcrumbs-aside">
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="primary-data-raw">Primary Data: Raw</h1>
<p>These should be exclusively the primary "big data" generated by sequencers, genotypers, light-sheet, EM etc.<br />
In other words, this data cannot be regenerated without repeating the initial experiment.<br />
&nbsp;<br />
<strong>This - and ONLY this data - will be backed up in duplicate copies following our <a href="https://github.com/neurobioinfo/Neuro_Bioinfo_Core.Nextcloud/wiki/Policy:-Redundant-Backups">backup policy</a>.</strong><br />
&nbsp;<br />
NOTE: this platform is not intended to support the very large amounts of fundamental imaging (human MRI/PET) that Neuro PIs currently store on the BIC server.<br />
&nbsp;<br />
If you wish this data to be backed up, its parent directory <strong>must</strong> be named "raw.data" (lowercase), found within your group's main data folder, i.e. the one named [PIname]Lab, with a folder-and-arrow icon 
(<img src="https://github.com/neurobioinfo/Neuro_Bioinfo_Core.Nextcloud/assets/35270039/6f51436c-75b1-49f0-9aae-25e02a51a69e" width=25>).<br />
&nbsp;<br />
Within this "raw.data" folder, your data may have any structure you like. However it is good practice to have directory structures that are as informative as possible. For example this might include:
* Having sub-folders by data type, e.g. genotyping, dnaseq, bulk Rnaseq, single-cell RNAseq, ATAC-seq, light-sheet, EM, microscopy etc.
* Having sub-folders by project name
&nbsp;  </p>
<p>Common raw data file types include: fastq/bam/cram (for sequence data), jpg/idat/txt (for genotyping data)</p>
<h1 id="secondary-and-tertiary-data-analyzed">Secondary and Tertiary Data: Analyzed</h1>
<p>These are the data generated from computationally-intensive analyses of your primary data.<br />
Users should apply discretion when deciding whether or not to store non-primary data on the Neuro_Bioinfo_Core storage platform.<br />
&nbsp;<br />
Reasons to store non-primary data include:<br />
<em> It is the product of very large amounts of computation and would be very costly/time-consuming to replace. (e.g. joint-called vcf for a large cohort)<br />
</em> It is a valuable project deliverable in itself<br />
<em> It is a common input for many downstream analyses (e.g. RNAseq gene count matrices)<br />
</em> You are required to keep it by university policy/publication requirement/grant mandate etc.  </p>
<p>We strongly discourage users from simply stockpiling all the files generated by their analyses with no curation.<br />
Practices to avoid include:
* Preserving intermediary files, i.e. those used only to generate final outputs. e.g.:
    * Alignment bam files produced by post-processing steps in dnaseq/rnaseq analyses
    * Single-sample dnaseq variant vcf files
    * Genotype files in various stages of filtration
* Preserving files which are easy/quick/inexpensive to regenerate
* Preserving extremely verbose analysis logs, core dumps, or files of unknown use/provenance</p>
<h1 id="personal-or-non-data-files">Personal or Non-Data files</h1>
<ul>
<li>Should ideally be stored outside of your group's main data folder, i.e. not in the one named [PIname]Lab, with a folder-and-arrow icon 
(<img src="https://github.com/neurobioinfo/Neuro_Bioinfo_Core.Nextcloud/assets/35270039/6f51436c-75b1-49f0-9aae-25e02a51a69e" width=25>).  </li>
<li>You can create any number of private folders in Nextcloud. You can name and organize them any way you like.  </li>
</ul>
<h1 id="common-considerations">Common Considerations</h1>
<ul>
<li>Data compression should be used whenever possible to save space. In particular, large plain-text formats such as fastq should be compressed with a utility like gzip.</li>
<li>Large collections of small files should be packaged whenever possible into archive formats such as tar/dar or zip.  </li>
<li>Data stored on this platform are not intended to be accessed frequently. Consider using a resource like the Digital Research Alliance of Canada (DRAC) server /project space for files that your group will re-use often.</li>
<li>Sharing files with external users via this platform is possible, but not optimal for large datasets. If you wish to share large amounts of data to a user outside this platform, please contact neurobioinfo@mcgill.ca for assistance.</li>
</ul>
<h1 id="additional-resources">Additional Resources</h1>
<p><a href="https://alliancecan.ca/sites/default/files/2022-05/Portage_Webinar_2020Sept16_SomeRDMBestPractices.pdf">Digital Research Alliance of Canada - Data Management Best Practices</a><br />
<a href="https://alliancecan.ca/sites/default/files/2022-05/Portage_Webinar_2020Sept29_BeginningWithTheEndInMind.pdf">Digital Research Alliance of Canada - Building Documentation and Metadata</a></p>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../redundant_backups/" class="btn btn-neutral float-left" title="Redundant Backups"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../download_data/" class="btn btn-neutral float-right" title="Download">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
    
      <span><a href="../redundant_backups/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../download_data/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "..";</script>
    <script src="../js/theme_extra.js"></script>
    <script src="../js/theme.js"></script>
      <script src="../search/main.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
